airflow/task/task_runner/standard_task_runner.py

```diff
@@ -42,6 +43,7 @@ def start(self):
         if CAN_FORK and not self.run_as_user:
             self.process = self._start_by_fork()
         else:
+            get_listener_manager().hook.on_starting(component=self)
```

Comment:
Hmmm, this one isn't "right", Calling it here puts it in the wrong process -- or at least the active memory of the process is about to get replace by the `exec` syscall that this path takes. And this will never get the before_stop called either.

Reply:
Moved the notification to `task command`.